key-share/src/lib.rs

```diff
@@ -88,6 +88,21 @@ use serde_with::As;
 ///   [extended_public_key](DirtyCoreKeyShare::extended_public_key) method).
 ///   * Setting `chain_code` to `None` disables HD wallets support for the key
 /// * Convenient methods are provided such as [derive_child_public_key](DirtyCoreKeyShare::derive_child_public_key)
+///
+/// # Serialization format via `serde`
+/// We make our best effort to keep serialization format the same between the versions (even with breaking changes),
+/// and so far we've never introduced breaking change into the serialization format. This ensures that newer versions
+/// of library are able to deserialize the key shares produced by the old version version of the library.
+///
+/// It's unlikely, but at some point, we might introduce a breaking change into the serialization format. In this case,
+/// we'll announce it and publish the migration instructions.
+///
+/// Not every serde backend is supported. We strongly advise using either [`serde_json`](https://docs.rs/serde_json/),
+/// if verbose/human-readable format is needed, or [`ciborium`](https://docs.rs/ciborium/latest/ciborium/), if you'd
+/// like to opt for binary format. Other serialization backends are not tested and disadvised to use.
```

Comment:
Maybe add an explaination? "Not every serde backend supports some features we use to ensure backwards compatability, which you will only detect at runtime. We advise you to use either ..., which are tested to work with this library. If you use another serde backend, you should test that it works first.

Reply:
> If you use another serde backend, you should test that it works first

If it works today, it doesn't mean it'll work tomorrow. Bincode worked with out lib before v0.2

Reply:
I expanded this section